CMake: Heffte include directories not passed to targets

When building CabanaMD and including Cabana-Cajita's`FastFourierTransform.hpp`, which itself includes `heffte_fft3d.h`, a build error occurs.

Here is how I am building Cabana:
```cmake
cmake \
     -D CMAKE_BUILD_TYPE="Debug" \
     -D CMAKE_PREFIX_PATH="$KOKKOS_INSTALL_DIR;$HEFFTE_INSTALL_DIR" \
     -D CMAKE_INSTALL_PREFIX=$CABANA_INSTALL_DIR \
     -D Cabana_ENABLE_MPI=ON \
     -D Cabana_ENABLE_TESTING=ON \
     -D Cabana_ENABLE_EXAMPLES=ON \
     -D Cabana_ENABLE_CAJITA=ON \
     -D Cabana_ENABLE_HEFFTE=ON \
     .. ;
```
 and CabanaMD: 
```cmake
cmake -D CMAKE_PREFIX_PATH="$CABANA_INSTALL_DIR" \
    -D CabanaMD_ENABLE_Serial=OFF \
    -D CabanaMD_ENABLE_OpenMP=ON \
    -D CabanaMD_ENABLE_Cuda=OFF \
    -D CabanaMD_ENABLE_LongRange=ON \
    .. ;
```

#258 is in progress and will fix this, unless the cause is actually something in CabanaMD itself or an error in how I'm building.

Error prints: `Cajita_FastFourierTransform.hpp:20:26: fatal error: heffte_fft3d.h: No such file or directory`

--- a/cajita/src/CMakeLists.txt
+++ b/cajita/src/CMakeLists.txt
@@ -36,7 +36,7 @@ if(HYPRE_FOUND)
     )
 endif()
 
-if(HEFFTE_FOUND)
+if(Heffte_FOUND)
   list(APPEND HEADERS_PUBLIC
     Cajita_FastFourierTransform.hpp
     )

--- a/cajita/src/CMakeLists.txt
+++ b/cajita/src/CMakeLists.txt
@@ -55,6 +55,7 @@ endif()
 
 if(Heffte_FOUND)
   target_link_libraries(Cajita Heffte::Heffte)
+  target_include_directories(Cajita PUBLIC ${Heffte_INCLUDE_DIR})
 endif()
 
 target_include_directories(Cajita

--- a/cajita/src/CMakeLists.txt
+++ b/cajita/src/CMakeLists.txt
@@ -44,18 +44,17 @@ endif()
 
 add_library(Cajita ${SOURCES_PUBLIC})
 
-target_link_libraries(Cajita
+target_link_libraries(Cajita PUBLIC
   Kokkos::kokkos
   MPI::MPI_CXX
   )
 
 if(HYPRE_FOUND)
-  target_link_libraries(Cajita HYPRE::hypre)
+  target_link_libraries(Cajita PUBLIC HYPRE::hypre)
 endif()
 
 if(Heffte_FOUND)
-  target_link_libraries(Cajita Heffte::Heffte)
-  target_include_directories(Cajita PUBLIC ${Heffte_INCLUDE_DIR})
+  target_link_libraries(Cajita PUBLIC Heffte::Heffte)
 endif()
 
 target_include_directories(Cajita

